### PR TITLE
EDSC-2681: Adds Tree component

### PR DESCRIPTION
**Pull request by macrouch**

### CODE REVIEW

File: src/components/FormElement/FormElement.js [hunk 1]
```diff
@@ -187,18 +213,39 @@ export const FormElement = ({
...
   if (tagName === 'group') {
     return (
-      <Group {...defaultProps} model={model}>
+      <Group {...defaultProps}>
         {children}
       </Group>
     )
   }
+  if (tagName === 'tree') {
+    const cascade = getAttribute(attributes, 'cascade') || 'true'
```
Comment (macrouch) [inline src/components/FormElement/FormElement.js]: add useRef for these attributes

File: src/components/Tree/TreeItem.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    item.setExpanded(!isExpanded)
+    setIsExpanded(!isExpanded)
+  }
+
+  return (
+    <div
+      key={timeStamp}
+      style={{ marginLeft: level * 15 }}
+    >
+      { children.length > 0 && !isExpanded && (
```
Comment (macrouch) [inline src/components/Tree/TreeItem.js]: isParent ?

File: src/util/updateModel.js [hunk 3]
```diff
@@ -23,7 +24,21 @@ export const updateModel = (model, modelRef, newValue) => {
       return element.outerHTML
     })
     value.innerHTML = values.join('')
+  } else if (typeof newValue === 'object') {
```
Comment (macrouch) [inline src/util/updateModel.js]: do I need this?

File: src/components/Tree/Tree.js [hunk 4]
```diff
@@ -0,0 +1,162 @@
+/* eslint-disable react-hooks/exhaustive-deps */
```
Comment (trevorlang) [inline src/components/Tree/Tree.js:1]: What’s this preventing?

File: src/util/TreeNode.js [hunk 5]
```diff
@@ -0,0 +1,366 @@
+import murmurhash from 'murmurhash'
+
+import { getAttribute } from './getAttribute'
+import { getNodeValue } from './getNodeValue'
+
+export class TreeNode {
+  constructor(props) {
+    // properties based on props
```
Comment (trevorlang) [inline src/util/TreeNode.js:8]: Nit picky, but inconsistent capitalization in comments 😬 😂

**Review by trevorlang — APPROVED** [review]

File: src/components/Tree/Tree.js (hunk 4 above)
Comment (macrouch) [reply src/components/Tree/Tree.js:1]: The linter doesn't like the dependency arrays on useEffect how we are using them. I think I might end up disabling this rule project wide

**Pull request merged**